data-generation/src/main/scala/UpdateDistributions.scala

```diff
@@ -0,0 +1,183 @@
+   *
+   * @param sc           A SparkContext
+   * @param graph        A graph
+   * @param mode         Mode for distributing the weights based on vertex degree
+   * @param distribution Some probability distribution
+   * @param mu           Expected value
+   * @param sigma        Standard deviation
+   * @return */
+  def addVertexUpdateDistribution[VD, ED: ClassTag](sc: SparkContext, graph: Graph[VD, ED], mode: UpdateDistributionMode, distribution: DistributionType, mu: Int, sigma: Double): Graph[Int, ED] = {
+    val graphWithDegree = graph
+      .vertices
+      .zip(graph.ops.degrees)
+      .map(vertex => (vertex._1._1, vertex._2._2))
+      .collect()
+      .sortBy(vertexTuple => vertexTuple)(getVertexSorting(mode))
+
+    val updates = graph.mapVertices((_, _) => getDistributionDraw(distribution, mu, sigma))
+      .vertices.collect()
+      .sortBy(_._2)
+      .map(_._2)
+
+    val rdd = sc.parallelize(
+      graphWithDegree
+        .zip(updates)
+        .map(vertex => (vertex._1._1, vertex._2))
+    )
+
+    Graph[Int, ED](rdd, graph.edges)
+  }
+
+  /** Get a vertex ordering rule based on a distribution mode
+   *
+   * @param mode Indicates the mode of which the nodes are ordered by
+   * @return A Ordering function
+   * */
+  private def getVertexSorting(mode: UpdateDistributionMode): Ordering[(VertexId, Int)] = {
+    mode match {
+      case UpdateDistributionMode.Uniform => (x: (VertexId, Int), y: (VertexId, Int)) => x._1 compareTo y._1
+      case UpdateDistributionMode.PositiveCorrelation => (x: (VertexId, Int), y: (VertexId, Int)) => x._2 compareTo y._2
+      case UpdateDistributionMode.NegativeCorrelation => (x: (VertexId, Int), y: (VertexId, Int)) => y._2 compareTo x._2
+    }
+  }
+
+  /** Add an update count to all edges
+   *
+   * @param sc           A SparkContext
+   * @param graph        A graph where vertices have a update count as a Int
+   * @param mode         Mode for distributing the weights based on vertex degree
+   * @param distribution Some probability distribution
+   * @param mu           Expected value
+   * @param sigma        Standard deviation
+   * @return */
+  def addEdgeUpdateDistribution[VD, ED: ClassTag](sc: SparkContext, graph: Graph[Int, ED], mode: UpdateDistributionMode, distribution: DistributionType, mu: Int, sigma: Double): Graph[Int, Int] = {
+    val vertexUpdateHashMap = graph
+      .vertices
+      .collect()
+      .toMap
+
+    val graphWithDegree = graph
+      .mapEdges(edge => getVertexUpdateSum[ED](edge, vertexUpdateHashMap))
+      .edges
+      .collect()
+      .sortBy(edge => edge)(getEdgeSorting(mode))
+
+    val updates = graph
+      .mapEdges(_ => getDistributionDraw(distribution, mu, sigma))
+      .edges
+      .collect()
+      .sortBy(edge => edge.attr)
+      .map(edge => edge.attr)
+
+    val rdd = sc.parallelize(
+      graphWithDegree
+        .zip(updates)
+        .map(edgeTuple => Edge(edgeTuple._1.srcId, edgeTuple._1.dstId, edgeTuple._2))
```

Comment:
kunne brukt underscore her også, men ikke viktig

Reply:
```suggestion
        .map(Edge(_._1.srcId, _._1.dstId, _._2))
```
?
In that case, does not work 🤔 Skjønner ikke helt greia med _, får det til av og til hahaha

Reply:
Snodig. Kan diskutere det litt etterpå

Reply:
Jeg tror hvis man bruker underscore flere ganger så antar den at det er flere argumenter.
```
.map(x => x+1)
.map( _+1)

.map(x,y => x + y)
.map(_ + _) // her betyr hver underscore noe forskjellig

// så da ser kanskje desukringen så her ut
.map(Edge(_._1.srcId, _._1.dstId, _._2))
.map( x,y,z => Edge(x._1.srcId, y._1.dstId, z._2))
```